pkg/policy/api/rule_validation.go

```diff
@@ -93,6 +93,12 @@ func (pr PortRule) Validate() error {
 		}
 	}
 
+	// Validate L7 rules
+	if pr.Rules != nil {
+		if (pr.Rules.HTTP != nil) && (pr.Rules.Kafka != nil) {
```

Comment:
Factorize this validation code into a new `(pr *L7Rule) Validate()` method.

Reply:
In your new `L7Rule.Validate` method, also validate your new Kafka rules in depth. This is not done for HTTP rules, but that's not an excuse for not doing that for Kafka rules. Define a new `(kr *PortRuleKafka) Validate()` method. Iterate on the `Kafka` list and call that validate method, where you should validate that all the fields are valid regexps (calling Go's `regexp.CompilePOSIX` and checking the returned error should be sufficient).

Reply:
Added the interface, working on the additional in-depth rule checking part.